frontend/src/lib/components/TaxonomicFilter/taxonomicFilterLogic.tsx

```diff
@@ -70,12 +72,12 @@ export const taxonomicFilterLogic = kea<taxonomicFilterLogicType>({
         }),
     }),
 
-    reducers: ({ selectors }) => ({
+    reducers: ({ selectors, props }) => ({
         searchQuery: [
             '',
             {
                 setSearchQuery: (_, { searchQuery }) => searchQuery,
-                selectItem: () => '',
+                selectItem: (state) => (props.clearSearchOnSelection === false ? state : ''),
```

Comment:
Reducers should technically be pure functions. That means the same `state` + `payload` combo should always return the same value. Here we don't know if `props` has changed in between. Won't block for it, just remarking on the code smell :).